File: src/test/java/com/zepben/vertxutils/routing/handlers/params/PathParamsTest.java
/*
 * Copyright 2020 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package com.zepben.vertxutils.routing.handlers.params;

import com.google.common.collect.ImmutableMap;
import org.junit.jupiter.api.Test;

import static com.zepben.testutils.exception.ExpectException.expect;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class PathParamsTest {

    private final PathParamRule<String> p1 = PathParamRule.Companion.of("p1", ParamType.INSTANCE.getSTRING());
    private final PathParamRule<Integer> p2 = PathParamRule.Companion.of("p2", ParamType.INSTANCE.getINT());
    private final PathParamRule<Integer> p3 = PathParamRule.Companion.of("p3", ParamType.INSTANCE.getINT());
    private final PathParamRule<String> notRegistered = PathParamRule.Companion.of("none", ParamType.INSTANCE.getSTRING());

    private final PathParams params = new PathParams(ImmutableMap.of(p1.getName(), "aString", p2.getName(), 4));

    @Test
    public void get() {
        assertThat(params.get(p1), is("aString"));
        assertThat(params.get(p2), is(4));
        expect(() -> params.get(p3)).toThrow(IllegalArgumentException.class);
        expect(() -> params.get(notRegistered)).toThrow(IllegalArgumentException.class);
    }

    @Test
    public void exists() {
        assertThat(params.exists(p1), is(true));
        assertThat(params.exists(p2), is(true));
        assertThat(params.exists(p3), is(false));
        assertThat(params.exists(notRegistered), is(false));
    }
}

File: src/test/java/com/zepben/vertxutils/routing/handlers/params/PathParamsTest.kt
/*
 * Copyright 2026 Zeppelin Bend Pty Ltd
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package com.zepben.vertxutils.routing.handlers.params

import com.zepben.testutils.exception.ExpectException.Companion.expect
import com.zepben.testutils.junit.SystemLogExtension
import com.zepben.vertxutils.routing.handlers.params.ParamType.INT
import com.zepben.vertxutils.routing.handlers.params.ParamType.STRING
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.RegisterExtension

class PathParamsTest {

    companion object {

        @JvmField
        @RegisterExtension
        val systemOut: SystemLogExtension = SystemLogExtension.SYSTEM_OUT.captureLog().muteOnSuccess()

    }

    private val p1 = PathParamRule.of("p1", STRING)
    private val p2 = PathParamRule.of("p2", INT)
    private val p3 = PathParamRule.of("p3", INT)
    private val notRegistered = PathParamRule.of("none", STRING)

    private val params = PathParams(mapOf(p1.name to "aString", p2.name to 4))

    @Test
    fun get() {
        assertThat(params[p1], equalTo("aString"))
        assertThat(params[p2], equalTo(4))
        expect { params[p3] }.toThrow<IllegalArgumentException>()
        expect { params[notRegistered] }.toThrow<IllegalArgumentException>()
    }

    @Test
    fun contains() {
        assertThat(p1 in params, equalTo(true))
        assertThat(p2 in params, equalTo(true))
        assertThat(p3 in params, equalTo(false))
        assertThat(notRegistered in params, equalTo(false))
    }
}
